src/Runner.Listener/Runner.cs

```diff
@@ -458,6 +458,35 @@ private async Task<int> RunAsync(RunnerSettings settings, bool runOnce = false)
                                     }
                                 }
                             }
+                            // Broker flow
+                            else if (string.Equals(message.MessageType, JobRequestMessageTypes.RunnerJobRequest, StringComparison.OrdinalIgnoreCase))
+                            {
+                                if (autoUpdateInProgress || runOnceJobReceived)
+                                {
+                                    skipMessageDeletion = true;
+                                    Trace.Info($"Skip message deletion for job request message '{message.MessageId}'.");
+                                }
+                                else
+                                {
+                                    var messageRef = StringUtil.ConvertFromJson<RunnerJobRequestRef>(message.Body);
+
+                                    // Create connection
+                                    var credMgr = HostContext.GetService<ICredentialManager>();
+                                    var creds = credMgr.LoadCredentials();
+
+                                    // todo: add retries https://github.com/github/actions-broker/issues/49
+                                    var runServer = HostContext.CreateService<IRunServer>();
+                                    await runServer.ConnectAsync(new Uri(settings.ServerUrl), creds);
```

Comment:
we are creating a new connection on every message.

Reply:
Well, we won't when we switch over to the run service 😄

Reply:
I am fine with the change now, but long term we might want to figure out all the URLs we need and where we get them. ex: broker URL, admin URL, runService URL.